pkg/controller/beat/common/driver.go

```diff
@@ -99,6 +103,9 @@ func Reconcile(
 
 	podTemplate, err := buildPodTemplate(params, defaultImage, configHash)
 	if err != nil {
+		if errors.Is(err, beat_stackmon.ErrMonitoringClusterUUIDUnavailable) {
+			results.WithReconciliationState(reconciler.Requeue.WithReason("ElasticsearchRef UUID unavailable while configuring beats stack monitoring")).WithResult(reconcile.Result{Requeue: true, RequeueAfter: 10 * time.Second})
```

Comment:
```suggestion
			results.WithReconciliationState(reconciler.RequeueAfter(10 * time.Second).WithReason("ElasticsearchRef UUID unavailable while configuring Beats stack monitoring"))
```

Reason is currently not used in Beats controller, but I guess it does not hurt either.